conda_lock/conda_lock.py

```diff
@@ -173,42 +186,29 @@ def do_conda_install(conda: PathLike, prefix: str, name: str, file: str) -> None
         args.extend(shlex.split(conda_flags))
 
     logging.debug("$MAMBA_ROOT_PREFIX: %s", os.environ.get("MAMBA_ROOT_PREFIX"))
-    proc = subprocess.run(
+
+    with subprocess.Popen(
         args,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
-        encoding="utf8",
-    )
-    logging.debug("install process: %s", proc)
-
-    def print_proc(proc):
-        print(f"    Command: {proc.args}")
-        if proc.stdout:
-            print(f"    STDOUT:\n{proc.stdout}")
-        if proc.stderr:
-            print(f"    STDERR:\n{proc.stderr}")
-
-    try:
-        proc.check_returncode()
-    except subprocess.CalledProcessError:
+        bufsize=1,
+        universal_newlines=True,
+    ) as p:
+        if p.stdout:
+            for line in _process_stdout(p.stdout):
+                logging.info(line)
+
+        if p.stderr:
+            for line in p.stderr:
+                logging.error(line if not line.endswith("\n") else line[:-1])
```

Comment:
can probably just do?

```suggestion
                logging.error(line.rstrip())
```

Reply:
i actually didn't know about this, thanks!